uberenv.py

```diff
@@ -493,32 +522,51 @@ def install(self):
         # note: this assumes package extends python when +python
         # this may fail general cases
         if self.opts["install"] and "+python" in full_spec:
-            activate_cmd = "spack/bin/spack activate " + self.pkg_name
+            activate_cmd = "spack/bin/spack activate " + self.pkg_name + self.opts["spec"]
             sexe(activate_cmd, echo=True)
         # if user opt'd for an install, we want to symlink the final
         # install to an easy place:
-        if self.opts["install"]:
-            pkg_path = self.find_spack_pkg_path(self.pkg_name)
+        if self.opts["install"] or "use_install" in self.opts:
+            pkg_path = self.find_spack_pkg_path(self.pkg_name,self.opts["spec"])
```

Comment:
Could use hash here: package already installed at this point, so hash is the robust way.

Reply:
Done